beluga/include/beluga/algorithm/estimation.hpp

```diff
@@ -102,6 +104,68 @@ Sophus::Matrix2<Scalar> calculate_covariance(Range&& range, const Sophus::Vector
       mean);
 }
 
+/// Returns a pair consisting of the estimated mean pose and its covariance.
+/**
+ * Given a range of poses, computes the estimated pose by averaging the translation
+ * and rotation parts.
+ * Computes the covariance matrix of the translation and rotation parts to create a 6x6 covariance matrix.
+ *
+ * \tparam Poses A [sized range](https://en.cppreference.com/w/cpp/ranges/sized_range) type whose
+ *  value type is `Sophus::SE3<Scalar>`.
+ * \tparam Weights A [sized range](https://en.cppreference.com/w/cpp/ranges/sized_range) type whose
+ *  value type is `Scalar`.
+ * \tparam Pose The pose value type of the given range.
+ * \tparam Scalar A scalar type, e.g. double or float.
+ * \param poses 3D poses to estimate mean and covariances from.
+ * \param weights Weights for the poses with matching indices.
+ * \return The estimated pose and its 6x6 covariance matrix.
+ */
+template <
+    class Poses,
+    class Weights,
+    class Pose = ranges::range_value_t<Poses>,
+    class Scalar = typename Pose::Scalar,
+    typename = std::enable_if_t<std::is_same_v<Pose, typename Sophus::SE3<Scalar>>>>
+std::pair<Sophus::SE3<Scalar>, Sophus::Matrix6<Scalar>> estimate(Poses&& poses, Weights&& weights) {
+  assert(std::size(poses) == std::size(weights));
+  assert(std::size(poses) > 0);
+  auto translation_view = poses | ranges::views::transform([](const auto& pose) { return pose.translation(); });
+  auto poses_view = poses | ranges::views::common;
+  auto weights_view = weights | ranges::views::common;
+  const auto weights_sum = std::accumulate(weights_view.begin(), weights_view.end(), 0.0);
+  auto normalized_weights_view =
+      weights_view | ranges::views::transform([weights_sum](const auto& weight) { return weight / weights_sum; });
+
+  const auto weighted_translation = [](const Pose& pose, const auto& weight) { return pose.translation() * weight; };
+
+  const auto weighted_rotation_vector = [](const Pose& pose, const auto& weight) {
+    return (pose.so3().log() * weight).eval();
+  };
+
+  const Sophus::Vector3<Scalar> mean_rotation_vector = std::transform_reduce(
+      poses_view.begin(), poses_view.end(), normalized_weights_view.begin(), Sophus::Vector3<Scalar>::Zero().eval(),
+      std::plus{}, weighted_rotation_vector);
```

Comment:
Consider using the lambda directly in the function call:

```suggestion
  const Sophus::Vector3<Scalar> mean_rotation_vector = std::transform_reduce(
      poses_view.begin(), //
      poses_view.end(), //
      normalized_weights_view.begin(), //
      Sophus::Vector3<Scalar>::Zero().eval(), //
      std::plus{}, //
      [](const Pose& pose, auto weight) { return (pose.so3().log() * weight).eval(); });
```

Same elsewhere.